test/gtest/adaptiveavgpool.hpp

```diff
@@ -0,0 +1,377 @@
+        {
+            D = 1;
+            H = in_grad_dim[2];
+            W = 1;
+
+            OD = 1;
+            OH = out_grad_dim[0];
+            OW = 1;
+            out_grad_dim_final.push_back(OH);
+        }
+        else if(in_grad_dim.size() == 4)
+        {
+            D = 1;
+            H = in_grad_dim[2];
+            W = in_grad_dim[3];
+
+            OD = 1;
+            OH = out_grad_dim[0];
+            OW = out_grad_dim[1];
+            out_grad_dim_final.push_back(OH);
+            out_grad_dim_final.push_back(OW);
+        }
+        else if(in_grad_dim.size() == 5)
+        {
+            D = in_grad_dim[2];
+            H = in_grad_dim[3];
+            W = in_grad_dim[4];
+
+            OD = out_grad_dim[0];
+            OH = out_grad_dim[1];
+            OW = out_grad_dim[2];
+            out_grad_dim_final.push_back(OD);
+            out_grad_dim_final.push_back(OH);
+            out_grad_dim_final.push_back(OW);
+        }
+        auto out_grad_strides = adaptiveavgpool_config.ComputeStrides(out_grad_dim_final);
+
+        auto gen_output_grad_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-10.0f), static_cast<T>(10.0f));
+        };
+        output_grad =
+            tensor<T>{out_grad_dim_final, out_grad_strides}.generate(gen_output_grad_value);
+
+        input_grad = tensor<T>{in_grad_dim};
+        std::fill(input_grad.begin(), input_grad.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_input_grad = tensor<T>{in_grad_dim};
+        std::fill(
+            ref_input_grad.begin(), ref_input_grad.end(), std::numeric_limits<T>::quiet_NaN());
+
+        output_grad_dev = handle.Write(output_grad.data);
+        input_grad_dev  = handle.Write(input_grad.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        miopenStatus_t status;
+
+        auto dims = input_grad.desc.GetNumDims();
+        if(dims == 3)
+        {
+            cpu_adaptiveavgpool_backward_1d(output_grad, ref_input_grad, N, C, H, OH);
+        }
+        else if(dims == 4)
+        {
+            cpu_adaptiveavgpool_backward_2d(output_grad, ref_input_grad, N, C, H, W, OH, OW);
+        }
+        else if(dims == 5)
+        {
+            cpu_adaptiveavgpool_backward_3d<T>(
+                output_grad, ref_input_grad, N, C, D, H, W, OD, OH, OW);
+        }
+        status = miopen::adaptiveavgpool::AdaptiveAvgPoolBackward(
+            handle, output_grad.desc, output_grad_dev.get(), input_grad.desc, input_grad_dev.get());
+
+        ASSERT_EQ(status, miopenStatusSuccess);
+
+        input_grad.data = handle.Read<T>(input_grad_dev, input_grad.data.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<T>::epsilon();
+        auto error       = miopen::rms_range(ref_input_grad, input_grad);
+        ASSERT_EQ(miopen::range_distance(ref_input_grad), miopen::range_distance(input_grad));
+        EXPECT_LT(error, threshold * 10);
```

Comment:
You should print something like this
```suggestion
        EXPECT_LT(error, threshold * 10) << "Error backward Input Gradient beyond 10xthreshold : " << error << " Tolerance: " << threshold * 10;
```

Reply:
Resolved